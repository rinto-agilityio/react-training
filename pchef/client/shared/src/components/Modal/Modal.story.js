// Libs
import React, { Component } from 'react'
import { Text } from 'react-native'
import { storiesOf } from '@storybook/react'

// Components
import Modal from '.'
import Button from '../Button'

// Helper
import { wInfo } from '../../../.storybook/utils';

type Props = {
  children?: React.Node,
  closeModal?: () => void,
}

const BaseModal = ({
  closeModal = () => { },
}: Props) => (
  <Modal
    onDismiss={closeModal}
    visible
    onSubmit={closeModal}
    title="Modal"
  >
    <Text>Modal content</Text>
  </Modal>
)

const ModalWithoutTitle = ({
  closeModal = () => { },
}: Prop) => (
  <Modal
    onDismiss={closeModal}
    visible
    onSubmit={closeModal}
  >
    <Text>Modal content</Text>
  </Modal>
)

const CustomModal = ({
  closeModal = () => { },
}: Props) => (
  <Modal
    onDismiss={closeModal}
    visible
    onSubmit={closeModal}
    dismissBtn
  >
    <Text>Modal content</Text>
  </Modal>
)

class StorybookModal extends Component<Props> {
  state = {
    openModal: false,
  }

  openModal = () => {
    this.setState({ openModal: true })
  }

  closeModal = () => {
    this.setState({ openModal: false })
  }

  render() {
    const { children } = this.props
    const { openModal } = this.state
    return (
      <>
        <Button
          title="Click me"
          onPress={this.openModal}
          buttonStyle={{
            width: 100,
          }}
        />
        {
          openModal &&
          React.cloneElement(children, {
            closeModal: this.closeModal,
          }) }
      </>
    )
  }
}

storiesOf('Modal', module)
  .addDecorator(wInfo())
  .add('Have title', () => (
    <StorybookModal>
      <BaseModal />
    </StorybookModal>
  ))
  .add('No title', () => (
    <StorybookModal>
      <ModalWithoutTitle />
    </StorybookModal>
  ))
  .add('Have dismiss button', () => (
    <StorybookModal>
      <CustomModal />
    </StorybookModal>
  ))
