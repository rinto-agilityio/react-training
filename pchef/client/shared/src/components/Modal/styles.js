// Libs
import { StyleSheet } from 'react-native'

// Themes
import { METRICS } from '../../themes'

const styles = StyleSheet.create({
  // Wrapper
  wrapper: {
    display: 'block',
    position: 'fixed',
    top: 0,
    left: 0,
    bottom: 0,
    right: 0,
    zIndex: METRICS.modalZindex,
  },

  // Dialog
  // Width full container for small/medium dialog
  largeDialog: {
    width: METRICS.mediumContent,
    alignSelf: 'center',
  },

  // Button
  btnModal: {
    marginRight: METRICS.mediumMargin,
  },
})

export default styles
