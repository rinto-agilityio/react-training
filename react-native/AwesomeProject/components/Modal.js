import React from 'react'
import { Modal, Text, TouchableHighlight, View } from 'react-native'

export default class ModalExample extends React.Component {
  state = {
    modalVisible: false
  }

  setModalVisible(visible) {
    this.setState({ modalVisible: visible })
  }

  render() {
    return (
      <View style={{marginTop: 22}}>
        <Modal
          animationType="slide"
          transparent={false}
          visible={this.state.modalVisible}
          onRequestClose={
            () => { alert("Modal has been close!")}
          }
        >
          <View style={{marginTop: 22}}>
            <View>
              <Text>Hellow World!</Text>

              <TouchableHighlight
                onPress={() => {
                  this.setModalVisible(!this.state.modalVisible)
                }}
              >
                <Text>Hide Modal</Text>
              </TouchableHighlight>
            </View>
          </View>
        </Modal>

        <TouchableHighlight
          onPress={() => {
            this.setModalVisible(true)
          }}
        >
          <Text>Show Modal</Text>
        </TouchableHighlight>
      </View>
    )
  }
}
