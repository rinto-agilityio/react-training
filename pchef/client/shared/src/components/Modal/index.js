// Libs
import * as React from 'react'
import { Dialog } from 'react-native-paper'

// Styles
import styles from './styles'

// Components
import Button from '../Button'

type Props = {
  size: string,
  title?: string,
  dismissBtn?: boolean,
  children: React.Node,
  visible?: boolean,
  customDialog?: {} | Array<{}>,
  onDismiss?: () => void,
  onSubmit?: () => void,
}

const Modal = ({
  size = 'medium',
  title,
  dismissBtn,
  children,
  onSubmit,
  visible,
  onDismiss,
  customDialog,
}: Props) => (
  <Dialog
    visible={visible}
    onDismiss={onDismiss}
    style={[styles.dialog, styles[`${size}Dialog`], customDialog]}
  >
    {title ? (
      <Dialog.Title
        style={[styles.titleModal, styles[`${size}TitleModal`]]}
        accessibilityRole="heading"
      >
        {title}
      </Dialog.Title>
    ) : null }
    <Dialog.Content style={[styles.content, styles[`${size}Content`]]}>
      {children}
    </Dialog.Content>
    <Dialog.Actions>
      {dismissBtn && (
        <Button
          onPress={onDismiss}
          title="Cancel"
          buttonStyle={[styles.btnModal, styles[`${size}btnModal`]]}
        />
      )}
      <Button
        onPress={onSubmit}
        title="Done"
        buttonStyle={[styles.btnModal, styles[`${size}btnModal`]]}
      />
    </Dialog.Actions>
  </Dialog>
)

Modal.defaultProps = {
  dismissBtn: false,
  title: '',
  visible: false,
  customDialog: {},
  onSubmit: () => {},
  onDismiss: () => {},
}

export default Modal
