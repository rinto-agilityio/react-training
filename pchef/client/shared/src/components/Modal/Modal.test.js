// Components
import Modal from '.'

describe('Modal', () => {
  let props = {
    dismissBtn: true,
  }

  it('Renders correctly modal commponent', () => {
    const component = renderer.create(<Modal {...props} />).toJSON()
    expect(component).toMatchSnapshot()
  })

  it('Render wish list component with size medium', () => {
    props = {
      ...props,
      size: 'medium',
    }
    const component = renderer.create(<Modal {...props} />).toJSON()
    expect(component).toMatchSnapshot()
  })

  it('Render wish list component with titel', () => {
    props = {
      ...props,
      title: 'Modal',
    }
    const component = renderer.create(<Modal {...props} />).toJSON()
    expect(component).toMatchSnapshot()
  })

  it('Render wish list component with defaultProps function', () => {
    Modal.defaultProps.onSubmit()
    expect(Modal.defaultProps.onSubmit).toBeDefined()
  })

  it('Render wish component with defaultProps function', () => {
    Modal.defaultProps.onDismiss()
    expect(Modal.defaultProps.onDismiss).toBeDefined()
  })
})
