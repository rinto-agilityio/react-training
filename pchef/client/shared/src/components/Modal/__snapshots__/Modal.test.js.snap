// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Render wish list component with size medium 1`] = `
<div
  className="css-view-1dbjc4n r-bottom-1p0dtai r-display-1adg3ll r-left-1d2f490 r-position-1xcajam r-right-zchlnj r-top-ipm5af r-zIndex-1sg8ghl"
/>
`;

exports[`Modal Render wish list component with titel 1`] = `
<div
  className="css-view-1dbjc4n r-bottom-1p0dtai r-display-1adg3ll r-left-1d2f490 r-position-1xcajam r-right-zchlnj r-top-ipm5af r-zIndex-1sg8ghl"
/>
`;

exports[`Modal Renders correctly modal commponent 1`] = `
<div
  className="css-view-1dbjc4n r-bottom-1p0dtai r-display-1adg3ll r-left-1d2f490 r-position-1xcajam r-right-zchlnj r-top-ipm5af r-zIndex-1sg8ghl"
/>
`;
